### ISSUE TITLE
No issue

### ISSUE THREAD

**Issue by cr9c1** [cr9c1/softether-4.21.9613-tls1.2-ecdhe#1]
@maurerr Were you able to get the source and binaries I put up?

**Comment by maurerr** [comment]
hi
i was able to get the binaries, install the windows client and compile the ubuntu binary but it connected with rc4-md5 client and i didn't have time to debug it further.
Do I have to configure a parameter in the comfig file ?

thanks

**Comment by cr9c1** [comment]
Thanks for bringing that to my attention. I had missed updating a header file with the default cipher. It has been fixed, and the source and windows binaries have been updated.  In the windows binary, the vpn server build is actually a collection of tools including the server, and the server manager.  You can use vpncmd or the server manager to set the cipher on the build you downloaded, or are free to download the new source and binaries.  If you wish to use the source you've already downloaded, then just log into the server with vpncmd or install the windows server manager and select the cipher under the Encryption settings.

**Comment by maurerr** [comment]
awesome job.
it works !
i'll do some benchmarks tomorrow

**Comment by cr9c1** [comment]
Glad to hear it. I have a .deb package with a forking systemd script as well I've been working on, haven't finished it yet but I've got most of it done. Looking forward to the benchmarks.

**Comment by maurerr** [comment]
ok, here goes the benchmarking:
with dhe-rsa-aes-128-cbc-sha i could get to 80mbps with my celeron n2820 as server
and now with ecdhe-rsa-aes256-gcm-sha284 i can only get half - actually less than 40mbps.
I would be great if you could rebuild with full TLS_1.2 ciphers available so i can test:
https://github.com/ssllabs/research/wiki/SSL-and-TLS-Deployment-Best-Practices#23-use-secure-cipher-suites

**Comment by cr9c1** [comment]
Interesting results.  The control channel paste from openvpn looks correct, its using TLS 1.2, and using the correct cipher (per how I built it).

I am curious if you'd be willing to try something for me and re-benchmark? If you wouldn't mind modifying your server's vpn_config file (either through the windows client or Linux CLI, and I've found if done through linux cli you must stop the vpn server, make the change, and restart it) and change the dhparambits from 4096 to 2048, its under the declare ServerConfiguration block. Change that to 2048, restart the server, and try again.

I can't provide any real world testing above or near 100mbps right now. I am limited by a 10/100 router, but I noticed you are using a Celeron N2820, which doesn't contain the AES-NI instruction sets which speed up AES encryption. Unfortunately, neither am I, as I am using a 1st Gen i7-950 (I have 4 cores assigned to my ubuntu vpn server), but I am going to try some iperf testing over the vpn inside my local network and see what I can come up with. I'm just curious to know if the change from 4096 to 2048 on your end will make a difference.

I looked at the link you posted with the additional cipher suites. I'll work on adding some of them in to give us more of an option.  I will probably get it done later tonight, I'm at work 7-4 and won't have time to mess with it until around 8-9PM tonight (EST). Let me know what you find out with the change.

**Comment by maurerr** [comment]
Hi,
my testing env has a big flaw  - it uses a shared proxy and performance varies with user load so until i manage to test in a controlled environment please ignore my benchmarks

**Comment by cr9c1** [comment]
Alright, no problem. If you have a chance to try changing from 4096 to 2048, and doing another test, I'd be curious to see if there is a difference.

**Comment by maurerr** [comment]
hi again,
still no relevant benchmark data but i can report a first issue :)
No vpnazure connection.
i have a US vps with 20 random ports and I used vpnazure to connect to 443 behind a corporate fw and now i cannot

**Comment by cr9c1** [comment]
Thanks for the report. I'm not familiar with the vpnazure stuff.  I'll have to look into it, but no promises. I'm not much of a programmer, I know enough to be dangerous.

**Comment by cr9c1** [comment]
@maurerr  It looks like softether released a very updated version with a lot of the features this modified version I'm running has. You should give it a go and see how it works. On another note, yo had mentioned you were running a Celeron N2820 for your server. Are you running that with a solid state hard drive? Curious about how the performance is, as I've considered something such as the J3455 to replace my old power hungry i7 950..

**Comment by maurerr** [comment]
Yes. I've installed latest version and it fixes vpn azure. Thanks
About n2820 yes I'm running on an older ssd. 
Performance is not so great. N2820 is actually an atom. 
If you can afford it go for a lowered powered i3
